src/hierarchies/base_hierarchy.hpp

```diff
@@ -27,11 +28,39 @@
 //! likelihood (whose parameters are the state) or its marginal distribution.
 
 class BaseHierarchy {
+ protected:
+  // map: data_id -> datum
+  std::unordered_map<int, Eigen::VectorXd> cluster_data;
```

Comment:
Do we really need actual data in hierarchies? They are already stored in Algorithm after all, and they not shuffled around there. A simpler structure for IDs should be enough, unless you're planning to move them out of Algorithm entirely, in which case of course my point would be moot. Or maybe it's a modularity thing?

Reply:
Right, actually we only need the sufficient statistics as far as I understand now. 
Keeping track of data indexes would still be useful for debugging, so I was thinking of having an unordered set for those but then update it only if a DEBUG flag is defined. How does that sound?

Reply:
Sounds great!